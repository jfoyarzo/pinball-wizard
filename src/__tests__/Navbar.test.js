import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import MainNav from '../Components/Navbar';

describe('Navbar component tests', () => {
  it('renders correctly', () => {
    const navbar = render(
      <Router>
        <MainNav />
      </Router>,
    );
    expect(navbar).toMatchSnapshot();
  });

  it('displays the corresponding title in home', () => {
    const route = '/';
    const { getByText } = render(
      <MemoryRouter initialEntries={[route]}>
        <MainNav />
      </MemoryRouter>,
    );
    expect(getByText('most pinball machines')).toBeInTheDocument();
  });

  it('displays the corresponding title in details page', () => {
    const route = '/Portland';
    const { getByText } = render(
      <MemoryRouter initialEntries={[route]}>
        <MainNav />
      </MemoryRouter>,
    );
    expect(getByText('city view')).toBeInTheDocument();
  });
});
