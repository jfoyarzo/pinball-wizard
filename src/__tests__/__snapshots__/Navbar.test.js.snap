// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand navbar-light"
        style="height: 5vh;"
      >
        <div
          class="align-middle justify-content-between container"
        >
          <div
            class="d-flex align-items-center"
          >
            <svg
              fill="currentColor"
              height="1.6em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.6em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
              />
            </svg>
            <span
              class="fw-bold"
              style="font-size: 1.4em;"
            >
              2015
            </span>
          </div>
          <p
            class="fw-light"
            style="font-size: 1.2em;"
          >
            most pinball machines
          </p>
          <div>
            <svg
              class="me-3"
              fill="currentColor"
              height="1.6em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.6em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
              >
                <path
                  d="M0 0h24v24H0z"
                />
                <path
                  d="M0 0h24v24H0z"
                />
                <path
                  d="M0 0h24v24H0z"
                />
              </g>
              <path
                d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
              />
              <path
                d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
              />
            </svg>
            <svg
              fill="currentColor"
              height="1.6em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1.6em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand navbar-light"
      style="height: 5vh;"
    >
      <div
        class="align-middle justify-content-between container"
      >
        <div
          class="d-flex align-items-center"
        >
          <svg
            fill="currentColor"
            height="1.6em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.6em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
            />
          </svg>
          <span
            class="fw-bold"
            style="font-size: 1.4em;"
          >
            2015
          </span>
        </div>
        <p
          class="fw-light"
          style="font-size: 1.2em;"
        >
          most pinball machines
        </p>
        <div>
          <svg
            class="me-3"
            fill="currentColor"
            height="1.6em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.6em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M0 0h24v24H0z"
              />
            </g>
            <path
              d="M12 14c1.66 0 3-1.34 3-3V5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3z"
            />
            <path
              d="M17 11c0 2.76-2.24 5-5 5s-5-2.24-5-5H5c0 3.53 2.61 6.43 6 6.92V21h2v-3.08c3.39-.49 6-3.39 6-6.92h-2z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1.6em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1.6em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
